components/encoder/fake/encoder.go

```diff
@@ -111,19 +111,23 @@ func (e *fakeEncoder) Position(
 func (e *fakeEncoder) start(cancelCtx context.Context) {
 	e.activeBackgroundWorkers.Add(1)
 	utils.ManagedGo(func() {
+		lastTime := time.Now()
+		e.mu.RLock()
+		updateRate := e.updateRate
+		e.mu.RUnlock()
+		step := time.Duration(updateRate) * time.Millisecond
```

Comment:
This now ignores reconfiguration since Reconfigure won't affect the update rate in the for loop after it's started. 

However, I think this was the issues that gave you bad encoder performance for the software test you're writing, and I hate reconfigure anyway.  So help me get rid of it! Please make this an `always.Rebuild` resource and move the logic  from this driver's `Reconfigure` to the constructor function, and delete the  `Reconfigure` function.

Reply:
Will do